client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -300,40 +302,93 @@ export class ShippingBanner extends Component {
 
 			const wcsStoreUnsubscribe = wcsStore.subscribe( () => {
 				const latestState = wcsStore.getState();
-				const siteState =
-					latestState.extensions.woocommerce.woocommerceServices[
-						siteId
-					];
-				if ( siteState ) {
-					const orderState = siteState.shippingLabel[ orderId ];
-					if ( orderState ) {
-						if ( orderState.showPurchaseDialog ) {
-							if ( window.jQuery ) {
-								window
-									.jQuery( '#woocommerce-order-label' )
-									.show();
-							}
-							wcsStoreUnsubscribe();
+
+				const shippingLabelState = get(
+					latestState,
+					[
+						'extensions',
+						'woocommerce',
+						'woocommerceServices',
+						siteId,
+						'shippingLabel',
+						orderId,
+					],
+					null
+				);
+
+				const labelSettingsState = get(
+					latestState,
+					[
+						'extensions',
+						'woocommerce',
+						'woocommerceServices',
+						siteId,
+						'labelSettings',
+					],
+					null
+				);
+
+				const packageState = get(
+					latestState,
+					[
+						'extensions',
+						'woocommerce',
+						'woocommerceServices',
+						siteId,
+						'packages',
+					],
+					null
+				);
+
+				const locationsState = get( latestState, [
+					'extensions',
+					'woocommerce',
+					'sites',
+					siteId,
+					'data',
+					'locations',
+				] );
+
+				if (
+					shippingLabelState &&
+					labelSettingsState &&
+					labelSettingsState.meta &&
+					packageState &&
+					locationsState
+				) {
+					if (
+						shippingLabelState.loaded &&
+						labelSettingsState.meta.isLoaded &&
+						packageState.isLoaded &&
+						isArray( locationsState ) &&
+						! this.state.isWcsModalOpen
+					) {
+						if ( window.jQuery ) {
+							this.setState( { isWcsModalOpen: true } );
```

Comment:
When the "create shipping label" button is clicked, doesn't appear to be an immediate state that lets us know if the button has been clicked or not, so we store that info in the component's state to avoid opening the modal multiple times.